server/text_generation_server/utils/weights.py

```diff
@@ -198,20 +270,29 @@ def _has_tensor(self, tensor_name: str):
     def get_shape(self, tensor_name: str):
         return self._get_slice(tensor_name).get_shape()
 
-    def get_tensor(self, tensor_name: str, to_device=True):
+    def get_tensor(self, tensor_name: str, to_device=True, cast=True):
```

Comment:
Same not happy about this but this is required to not cast the fp8 scales to bfloat16 and keep them in fp32.

Reply:
Yeah, I had a discarded PR that did something similar. Maybe we can call it `to_dtype` for consistency with `to_device`?